include/phasar/PhasarLLVM/Utils/AnalysisPrinterBase.h

```diff
@@ -0,0 +1,44 @@
+#ifndef PHASAR_PHASARLLVM_UTILS_ANALYSISPRINTERBASE_H
+#define PHASAR_PHASARLLVM_UTILS_ANALYSISPRINTERBASE_H
+
+#include "llvm/Support/raw_ostream.h"
+
+namespace psr {
+
+template <typename AnalysisDomainTy> struct Warning {
```

Comment:
Would other/more categories make sense?

Reply:
The intuition here is that, static analysis only reports warnings due to the sound approximations. For example, an Error would mean that it would definetly occur at runtime and we can only report an Error if we use precise information in our analyses. IMHO, I don't think, there is a need for other categories yet but would definitely be a requirement if we use runtime information.